lib/ts/recipe/passwordless/utils.ts

```diff
@@ -79,15 +79,49 @@ export function validateAndNormaliseUserInput(
                   };
         return emailDelivery;
     }
-    if (config.contactMethod === "EMAIL") {
-        // TODO: to remove this
-        if (config.createAndSendCustomEmail === undefined) {
-            throw new Error("Please provide a callback function (createAndSendCustomEmail) to send emails. ");
+
+    function getSmsDeliveryConfig() {
+        let smsService = config.smsDelivery?.service;
+
+        let createAndSendCustomTextMessage =
+            config.contactMethod === "EMAIL" ? undefined : config.createAndSendCustomTextMessage;
+        /**
+         * following code is for backward compatibility.
+         * if user has not passed emailDelivery config, we
+         * use the createAndSendCustomTextMessage config. If the user
+         * has not passed even that config, we use the default
+         * createAndSendCustomTextMessage implementation
+         */
+        if (smsService === undefined) {
+            smsService = new BackwardCompatibilitySmsService(appInfo, createAndSendCustomTextMessage);
         }
+        let smsDelivery =
+            config.contactMethod === "EMAIL"
+                ? {
+                      service: smsService,
+                  }
+                : {
+                      ...config.smsDelivery,
+                      /**
+                       * if we do
+                       * let smsDelivery = {
+                       *    service: smsService,
+                       *    ...config.smsDelivery,
+                       * };
+                       *
+                       * and if the user has passed service as undefined,
+                       * it it again get set to undefined, so we
+                       * set service at the end
+                       */
+                      service: smsService,
+                  };
```

Comment:
```suggestion
            {
                      ...config.smsDelivery,
                      /**
                       * if we do
                       * let smsDelivery = {
                       *    service: smsService,
                       *    ...config.smsDelivery,
                       * };
                       *
                       * and if the user has passed service as undefined,
                       * it it again get set to undefined, so we
                       * set service at the end
                       */
                      service: smsService,
                  };
```

Reply:
The reason for not doing `config.contactMethod === "EMAIL" ?..` is that `smsService` is on the root level now. Same goes for emailsending service.

Reply:
done